chainer/training/extensions/parameter_statistics.py

```diff
@@ -103,15 +106,23 @@ def __call__(self, trainer):
                         # since the statistics function should make no
                         # assumption about the axes
                         params = getattr(param, attr_name).ravel()
-                        value = function(params)
+                        if self.check_nan_params and \
+                                cuda.get_array_module(params).isnan(params) \
+                                    .any():
+                            value = numpy.nan
+                        else:
+                            value = function(params)
                         key = self.report_key_template.format(
                             prefix=self._prefix + '/' if self._prefix else '',
                             link_name=link_name,
                             param_name=param_name,
                             attr_name=attr_name,
                             function_name=function_name
                         )
-                        if hasattr(value, '__iter__'):
+                        if hasattr(value, '__iter__') or \
+                                (isinstance(value,
```

Comment:
Why do you add this checking?

Reply:
Added it since cupy ndarrays don't have the `__iter__` attribute. Simplified it a bit though since the check is not needed for numpy.